src/GlobalConfig.cpp

```
@@ -0,0 +1,230 @@
+
+		// A missing section is not an error: every field then keeps its default. We hand back an empty object
+		// so the per-field readers run uniformly whether or not the section was present.
+		const rapidjson::Value& sectionOrEmpty(const rapidjson::Value& root, const char* key,
+			const rapidjson::Value& empty)
+		{
+			const auto* v = find(root, key);
+			return v && v->IsObject() ? *v : empty;
+		}
+	}
+
+	GlobalConfig parseConfigJson(std::string_view bytes, GlobalConfig base)
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)